packages/grid/data-grid/src/DataGrid.tsx

```diff
@@ -12,6 +12,7 @@ const FORCED_PROPS: Partial<GridComponentProps> = {
   pagination: true,
   apiRef: undefined,
   onRowsScrollEnd: undefined,
+  checkboxSelectionVisibleOnly: false,
```

Comment:
I would sort ASC, to make it easier to find and reduce likeness of conflicts (forced, prop-type, types, etc.)